### PR TITLE
#xngzj64v Ft-query-crud

### PR DESCRIPTION
**Pull request by niyongaboaristide17**
### What does this PR
Query CRUD api
### Description of Task to be completed?
- ReadOne
- ReadAll
- Create
- Update
- Delete
#### clone repo to locally
- git clone repo_link
#### install dependency
- npm install
#### Configure env
- create .env into the project folder
- copy .env.sample content to .env
- generate env variable content
##### create
- /api/v1/queries
##### read all
- /api/v1/queries
##### read one
- /api/v1/queries/{query_id}
##### update
- /api/v1/queries/{query_id}
##### delete
- /api/v1/queries/{query_id}

### What are the relevant pivotal tracker
- https://trello.com/b/xNgZJ64v/my-brand

### Screenshots
Screenshot from 2022-01-28 16-24-12
Screenshot from 2022-01-28 16-24-34
Screenshot from 2022-01-28 16-24-50
Screenshot from 2022-01-28 16-25-03
Screenshot from 2022-01-28 16-25-30

### CODE REVIEW

File: src/controllers/articleController.js [hunk 1]
```diff
@@ -64,25 +61,18 @@ export class ArticleController {
...
 
-        } catch {
-            res.status(404)
-            res.send({ error: "Article doesn't exist!" })
+
+        } catch (error) {
+            res.status(404).send({ error: "Article doesn't exist!" })
         }
     }
 }
```
Comment (ncutixavier) [inline src/controllers/articleController.js]: Add new line

File: src/controllers/queriesController.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+            res.send(newQuery)
+        } catch (error) {
+
+            res.status(404).send({ error: 'Something went wrong' })
+        }
+    }
+
+
+}
```
Comment (ncutixavier) [inline src/controllers/queriesController.js:77]: Add new line

File: src/models/query.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    sender: {
+        name: { type: String, required: true },
+        email: { type: String, required: true }
+    },
+    message: { type: String, required: true },
+    location: { type: String, required: true },
+    created_on: { type: Date, default: Date.now }
+})
+const Query = mongoose.model('Query', querySchema)
+export default Query
```
Comment (ncutixavier) [inline src/models/query.js:13]: Add new line

File: src/services/queryServices.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+        if (data.location) {
+            query.location = data.location
+        }
+
+        await query.save()
+        return query
+
+    }
+}
```
Comment (ncutixavier) [inline src/services/queryServices.js:44]: Add new line

**Review by ncutixavier — CHANGES_REQUESTED** [review]
Your PR title is not following naming convention

File: src/models/query.js (hunk 3 above)
Comment (niyongaboaristide17) [reply src/models/query.js:13]: What line?

File: src/services/queryServices.js (hunk 4 above)
Comment (niyongaboaristide17) [reply src/services/queryServices.js:44]: What line?

**Pull request merged**